digdag-standards/src/main/java/io/digdag/standards/command/ecs/EcsClientConfig.java

```diff
@@ -31,6 +32,12 @@ public EcsClientConfig(EcsClientConfigBuilder builder)
         this.memory = builder.getMemory();
         this.startedBy = builder.getStartedBy();
         this.assignPublicIp = builder.isAssignPublicIp();
+        this.placementStrategyType = builder.getPlacementStrategyType();
+        this.placementStrategyField = builder.getPlacementStrategyField();
+
+        if (!placementStrategyType.isPresent() && placementStrategyField.isPresent()) {
+            throw new ConfigException("PlacementStrategyField must be set with PlacementStrategyType");
+        }
```

Comment:
Don't we need to check the other case like `placementStrategyType.isPresent() && !placementStrategyField.isPresent()`?

Reply:
It depends what kind of `placementStrategyType` users use.
It's complicated a little, some `placementStrategyType` requires `placementStrategyField` but others does not.

But at least `placementStrategyField` only is invalid usage. That's the reason why I added the validation here.

Regarding more detail validations, I decided not to implement it here because it is complicated to cover all patterns as I told above. I assume digdag users who use EcsCommandExecutor have the knowledge about ECS placement strategy option.